MeshLib/MeshEditing/MeshRevision.cpp

```diff
@@ -110,28 +139,52 @@ MeshLib::Mesh* MeshRevision::subdivideMesh(const std::string &new_mesh_name) con
 	if (this->_mesh.getNElements() == 0)
 		return nullptr;
 
+	// data structures for the new mesh
 	std::vector<MeshLib::Node*> new_nodes = MeshLib::copyNodeVector(_mesh.getNodes());
 	std::vector<MeshLib::Element*> new_elements;
-
-	const std::vector<MeshLib::Element*> &elements(this->_mesh.getElements());
-	for (auto elem = elements.begin(); elem != elements.end(); ++elem)
-	{
-		ElementErrorCode e((*elem)->validate());
-		if (e[ElementErrorFlag::NonCoplanar])
+	MeshLib::Properties new_properties;
+	boost::optional<PropertyVector<int> &> new_material_vec(
+		new_properties.createNewPropertyVector<int>(
+			"MaterialIDs", MeshItemType::Cell, 1
+		)
+	);
```

Comment:
same comment as above: first check if the vector actually exists before creating a new one.